pkgs/servers/typedb/default.nix

```diff
@@ -0,0 +1,99 @@
+{ stdenv, lib, openjdk,typedbHome ? "~/.typedb_home", fetchzip, fetchurl}:
+
+let 
+  typedbVersion = "2.1.1";
+  typedbDirLinux = "typedb-all-linux-${typedbVersion}";
+  typedbDirMac = "typedb-all-mac-${typedbVersion}";
+  typedbDirWindows = "typedb-all-windows-${typedbVersion}";
+  typedbDir = if stdenv.hostPlatform.isWindows then typedbDirWindows
+              else if stdenv.isDarwin          then typedbDirMac
+                                               else typedbDirLinux;
+  linuxSrc = builtins.fetchTarball {
+    url = "https://github.com/vaticle/typedb/releases/download/2.1.1/typedb-all-linux-2.1.1.tar.gz";
+    sha256 = "15nwm2dr68p67c2xcqigs66gd679j1zr72gqv7qxgvflwyvyz8fb";
+  };
+  linuxSrcUrl = fetchurl {
+    url = "https://github.com/vaticle/typedb/releases/download/2.1.1/typedb-all-linux-2.1.1.tar.gz";
+    sha256 = "00wvkiahmrys32ckp6r59bbrzfd7jzsphniyjdj4p9bh7k339gn2";
+  };
+  windowsSrc = fetchzip {
+    url = "https://github.com/vaticle/typedb/releases/download/2.1.1/typedb-all-windows-2.1.1.zip";
+    sha256 = "0vd66gfshkg697z07nhy957mwqzlli4r4pmn67hx58n9mkg024kq";
+  };
+  macSrc = fetchzip {
+    url = "https://github.com/vaticle/typedb/releases/download/2.1.1/typedb-all-mac-2.1.1.zip";
+    sha256 = "16hlfy6kh2rnvcralz206q13mghb0rv8wazpg6q3h324p5rdys54";
+  };
+  srcFolder = if stdenv.hostPlatform.isWindows then windowsSrc
+              else if stdenv.isDarwin          then macSrc
+                                               else linuxSrcUrl ;
+  javaPatch = ''
+        20c20
+        < JAVA_BIN=java
+        ---
+        > JAVA_BIN=${openjdk}/bin/java
+        '';
```

Comment:
Indentation is a bit odd an there's some trailing whitespace. `nixpkgs-fmt` can help you out.

Reply:
done; did not know about `nixpkgs-fmt`, that helps a lot